### PR TITLE
Add 'Topic.get_iam_policy' API wrapper.

### PR DESCRIPTION
**Pull request by tseaver**
Returns an instance of `gcloud.pusub.iam.Policy`, parsed from the API response.

Toward #1073.

### CODE REVIEW

File: docs/pubsub-usage.rst [hunk 1]
```diff
@@ -66,6 +66,23 @@ Delete a topic:
...
+   >>> from gcloud import pubsub
+   >>> client = pubsub.Client()
+   >>> topic = client.topic('topic_name')
+   >>> policy = topic.get_iam_policy()  # API request
+   >>> policy.etag
+   'DEADBEEF'
+   >>> policy.owners
+   ['user:phred@example.com']
+   >>> policy.writers
+   ['systemAccount':abc-1234@systemaccounts.example.com']
```
Comment (dhermes) [inline docs/pubsub-usage.rst]: Invalid string

File: docs/pubsub-usage.rst [hunk 2]
```diff
@@ -66,6 +66,23 @@ Delete a topic:
    >>> topic = client.topic('topic_name')
    >>> topic.delete()  # API request
 
+Fetch the IAM policy for a topic:
```
Comment (dhermes) [inline docs/pubsub-usage.rst]: I thought @jonparrott said IAM was at the service level rather than the object level?

File: gcloud/pubsub/iam.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        :param email: E-mail for this particular user.
+
+        :rtype: string
+        :returns: A member string corresponding to the given user.
+        """
+        return 'user:%s' % (email,)
+
+    @staticmethod
+    def service_account(email):
+        """Factory method for a user member.
```
Comment (dhermes) [inline gcloud/pubsub/iam.py]: Copy-pasta docstring?

File: gcloud/pubsub/iam.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        :param domain: The domain for this member.
+
+        :rtype: string
+        :returns: A member string corresponding to the given domain.
+        """
+        return 'domain:%s' % (domain,)
+
+    @staticmethod
+    def all_users():
+        """Factory method for an member representing all users.
```
Comment (dhermes) [inline gcloud/pubsub/iam.py]: an member -> a member

File: gcloud/pubsub/iam.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+        :rtype: string
+        :returns: A member string representing all users.
+        """
+        return 'allUsers'
+
+    @staticmethod
+    def authenticated_users():
+        """Factory method for an member representing all authenticated users.
```
Comment (dhermes) [inline gcloud/pubsub/iam.py]: an member

File: gcloud/pubsub/iam.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+            resource['etag'] = self.etag
+
+        if self.version is not None:
+            resource['version'] = self.version
+
+        bindings = []
+
+        if self.owners:
+            bindings.append(
+                {'role': 'roles/owner', 'members': sorted(self.owners)})
```
Comment (dhermes) [inline gcloud/pubsub/iam.py]: Since `roles/owner` is used elsewhere, let's make it a constant. Ditto for the others.

File: gcloud/pubsub/iam.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+            role = binding['role']
+            members = set(binding['members'])
+            if role == 'roles/owner':
+                policy.owners = members
+            elif role == 'roles/writer':
+                policy.writers = members
+            elif role == 'roles/reader':
+                policy.readers = members
+            else:
+                raise ValueError('Unknown role: %s' % (role,))
```
Comment (dhermes) [inline gcloud/pubsub/iam.py]: Feel free to leave as-is, but `ValueError` and friends accept an arbitrary number of args so `ValueError('Unknown role', role)` would've been fine too.

File: docs/pubsub-usage.rst (hunk 2 above)
Comment (tseaver) [inline docs/pubsub-usage.rst]: https://cloud.google.com/pubsub/reference/rest/v1/projects.topics/getIamPolicy

File: docs/pubsub-usage.rst (hunk 1 above)
Comment (tseaver) [inline docs/pubsub-usage.rst]: baa4050

File: gcloud/pubsub/iam.py (hunk 3 above)
Comment (tseaver) [inline gcloud/pubsub/iam.py]: 9f2a604

File: gcloud/pubsub/iam.py (hunk 4 above)
Comment (tseaver) [inline gcloud/pubsub/iam.py]: 2a52215

File: gcloud/pubsub/iam.py (hunk 5 above)
Comment (tseaver) [inline gcloud/pubsub/iam.py]: 2a52215

File: gcloud/pubsub/iam.py (hunk 6 above)
Comment (tseaver) [inline gcloud/pubsub/iam.py]: ebf5051

**Comment by tseaver** [comment]
@dhermes I think I've addressed your issues.

**Comment by dhermes** [comment]
LGTM

**Pull request merged**